packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -1163,7 +1163,17 @@ class LocalizationsGenerator {
             }
             if (!pluralLogicArgs.containsKey(pluralCases[pluralCase])) {
               final String pluralPartExpression = generateVariables(pluralMessage);
-              pluralLogicArgs[pluralCases[pluralCase]!] = '      ${pluralCases[pluralCase]}: $pluralPartExpression,';
+              final String? transformedPluralCase = pluralCases[pluralCase];
+              if (transformedPluralCase == null) {
+                throw L10nParserException(
+                  'The plural case $pluralCase is not a valid plural case.',
```

Comment:
Shall we either document here or link to some documentation as to what the valid plural cases are?

Reply:
Fixed!